lib/manageiq/environment.rb

```diff
@@ -121,6 +121,46 @@ def self.bundler_version
       File.read(gemfile).match(/gem\s+['"]bundler['"],\s+['"](.+?)['"]/)[1]
     end
 
+    def self.warn_if_branch_out_of_date
+      return if %w(production test).include?(ENV['RAILS_ENV'].to_s)
+
+      Dir.chdir(APP_ROOT) do
+        system!("git fetch")
+
+        commits =
+          if upstream_tracking_branch.empty?
+            commits_behind_upstream
+          else
+            commits_behind_tracked_branch
+          end
+
+        if commits > 0
+          warn "\n== Your branch is #{commits} commits out of date, please pull or fetch + merge."
+        end
+      end
+    end
+
+    def self.upstream_tracking_branch
+      `git for-each-ref --format='%(upstream:short)' $(git symbolic-ref -q HEAD)`.strip
```

Comment:
The use of backtick hurts but I need the output and don't want to pull in a gem to do this when it's in plain ruby.